docs/design-notes/cql-extensions-internal.md

```diff
@@ -0,0 +1,38 @@
+# Scylla CQL extensions
+
+Scylla extends the CQL language to provide a few extra features. This document
+lists those extensions.
+
+## SCYLLA_TIMEUUID_LIST_INDEX function
+
+The `SCYLLA_TIMEUUID_LIST_INDEX` function allows using the raw (sstable) collection
+key for lists, which is a timeuuid, as "index" when performing a modification.
+
+The function is used to wrap the index part of a list assignment:
+
+    UPDATE ... SET list_column[SCYLLA_TIMEUUID_LIST_INDEX(:v1)] = :v2 ... 
+
+It is only used by sstableloader.
+
+## SCYLLA_COUNTER_SHARD_LIST function
+
+The `SCYLLA_COUNTER_SHARD_LIST` function allows setting a counter column value by 
+its internal representation.
+
+The function is used to wrap the value part of a counter assignment:
+
+    UPDATE ... SET counter_column = SCYLLA_COUNTER_SHARD_LIST(:v1) ... 
+
+It is only used by sstableloader.
+
+## SCYLLA_CLUSTERING_BOUND function
+
+The `SCYLLA_CLUSTERING_BOUND` function marks a query bound as being 
+a "raw" CQL clustering boundary, not a value range bound as in normal
+CQL.
+
+The function is used to wrap the clustering range values in a `WHERE` statement.
```

Comment:
Too vague.  How about a more definitive and precise language:

"This function can be used on the right-hand side of a [multicolumn comparison](https://github.com/scylladb/scylla/blob/2f3b265dac0acf3b2c243351bf466247ca9c7a0a/cql3/Cql.g#L1535), excluding the IN and LIKE operators. Its parameters are values corresponding to the clustering prefix specified on the left-hand side of that comparison."